From: Aristeu Rozanski <aris@redhat.com>
Date: Wed, 20 Aug 2008 13:17:01 -0400
Subject: [x86] nmi: introduce per-cpu wd_enabled
Message-id: 20080820171650.084476000@redhat.com
O-Subject: [RHEL5.3 PATCH 15/25] nmi: introduce per-cpu wd_enabled for i386
Bugzilla: 447618

https://bugzilla.redhat.com/show_bug.cgi?id=447618

introduce wd_enabled per-cpu variable: it'll determine if the NMI watchdog is
enabled on the current cpu

upstream: 09198e68501a7e34737cd9264d266f42429abcdc

diff --git a/arch/i386/kernel/nmi.c b/arch/i386/kernel/nmi.c
index c17f374..eb91d21 100644
--- a/arch/i386/kernel/nmi.c
+++ b/arch/i386/kernel/nmi.c
@@ -46,6 +46,8 @@ static unsigned int lapic_nmi_owner;
 #define LAPIC_NMI_WATCHDOG	(1<<0)
 #define LAPIC_NMI_RESERVED	(1<<1)
 
+static DEFINE_PER_CPU(short, wd_enabled);
+
 /* nmi_active:
  * +1: the lapic NMI watchdog is active, but can be disabled
  *  0: the lapic NMI watchdog has not been set up, and cannot
